<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
 <HEAD>
   <TITLE> [rabbitmq-discuss] ldap authentication
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:rabbitmq-discuss%40lists.rabbitmq.com?Subject=Re%3A%20%5Brabbitmq-discuss%5D%20ldap%20authentication&In-Reply-To=%3C4E793643.703%40gmail.com%3E">
   <META NAME="robots" CONTENT="index,nofollow">
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="015178.html">
   <LINK REL="Next"  HREF="015196.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[rabbitmq-discuss] ldap authentication</H1>
    <B>Nathan</B> 
    <A HREF="mailto:rabbitmq-discuss%40lists.rabbitmq.com?Subject=Re%3A%20%5Brabbitmq-discuss%5D%20ldap%20authentication&In-Reply-To=%3C4E793643.703%40gmail.com%3E"
       TITLE="[rabbitmq-discuss] ldap authentication">phenisha at gmail.com
       </A><BR>
    <I>Wed Sep 21 01:56:35 BST 2011</I>
    <P><UL>
        <LI>Previous message: <A HREF="015178.html">[rabbitmq-discuss] ldap authentication
</A></li>
        <LI>Next message: <A HREF="015196.html">[rabbitmq-discuss] ldap authentication
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#15195">[ date ]</a>
              <a href="thread.html#15195">[ thread ]</a>
              <a href="subject.html#15195">[ subject ]</a>
              <a href="author.html#15195">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>On 09/20/2011 10:33 AM, Matthias Radestock wrote:
&gt;<i> On 20/09/11 01:21, Nathan wrote:
</I>&gt;&gt;<i> when i change it back to rabbitmq_auth_backend_internal
</I>&gt;&gt;<i> none of my local users can authenticate?
</I>&gt;<i>
</I>&gt;<i> That needs to be rabbit_auth_backend_internal, no &quot;mq&quot;. That rule 
</I>&gt;<i> generally applies to the contents of auth_backends, since it refers to 
</I>&gt;<i> Erlang modules, which we usually name rabbit_*, whereas our Erlang 
</I>&gt;<i> applications (and hence config settings) are usually named rabbitmq_*. 
</I>&gt;<i> Except for the rabbit application, which is just called 'rabbit'. 
</I>&gt;<i> Confused yet?
</I>&gt;<i>
</I>&gt;<i> Matthias.
</I>
Ok now I have local auth back, thanks (and for the record I am confused)!

My current config looks like the following

[
  {rabbitmq_stomp, [{tcp_listeners, [6163]} ]},
  {rabbit, [{auth_backends, [rabbit_auth_backend_internal,  
rabbitmq_auth_backend_ldap]}]  },
  {rabbitmq_auth_backend_ldap,
   [ {servers,               [&quot;ldap.zozee.com&quot;]},
     {user_dn_pattern,       &quot;uid=${username},ou=people,dc=zozee,dc=com&quot;},
     {vhost_access_query,    
{exists,&quot;ou=${vhost},ou=people,dc=zozee,dc=com&quot;}},
     {resource_access_query,
       {for, [{resource, exchange,{for, [{permission, configure,
                                          { in_group, 
&quot;cn=zozee,ou=groups,dc=zozee,dc=com&quot; }
                                         },
                                         {permission, write, {constant, 
true}},
                                         {permission, read,  {constant, 
true}}
                                        ]}},
              {resource, queue, {constant, true}} ]}},
       {tag_queries,           [{administrator, {constant, false}}]},
       {use_ssl,               false},
       {port,                  389},
       {log,                   true} ] }
].


How ever I am still seeing errors in the log, as follows


=INFO REPORT==== 21-Sep-2011::10:54:01 ===
accepted TCP connection on [::]:5672 from 172.16.0.51:40465

=INFO REPORT==== 21-Sep-2011::10:54:01 ===
starting TCP connection &lt;0.1343.0&gt; from 172.16.0.51:40465

=ERROR REPORT==== 21-Sep-2011::10:54:04 ===
exception on TCP connection &lt;0.1343.0&gt; from 172.16.0.51:40465
{channel0_error,starting,
     {error,undef,'connection.start_ok',
         [{rabbitmq_auth_backend_ldap,check_user_login,
              [&lt;&lt;&quot;testguy&quot;&gt;&gt;,[{password,&lt;&lt;&quot;testldappw&quot;&gt;&gt;}]]},
          {rabbit_access_control,'-check_user_login/2-fun-0-',4},
          {lists,foldl,3},
          {rabbit_reader,auth_phase,2},
          {rabbit_reader,handle_method0,3},
          {rabbit_reader,handle_input,3},
          {rabbit_reader,recvloop,2},
          {rabbit_reader,start_connection,7}]}}

=INFO REPORT==== 21-Sep-2011::10:54:04 ===
closing TCP connection &lt;0.1343.0&gt; from 172.16.0.51:40465


Now, it looks like that it is trying to use the ldap module however i 
can not see any
connections on my ldap server from the rabbitmq server.
</PRE>







<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="015178.html">[rabbitmq-discuss] ldap authentication
</A></li>
	<LI>Next message: <A HREF="015196.html">[rabbitmq-discuss] ldap authentication
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#15195">[ date ]</a>
              <a href="thread.html#15195">[ thread ]</a>
              <a href="subject.html#15195">[ subject ]</a>
              <a href="author.html#15195">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">More information about the rabbitmq-discuss
mailing list</a><br>
</body></html>
